lib/pipeline.ts

```diff
@@ -338,17 +346,19 @@ Pipeline.prototype.exec = function (callback: CallbackFunction) {
 
   // In cluster mode, always load scripts before running the pipeline
   if (this.isCluster) {
-    return pMap(scripts, (script) => _this.redis.script("load", script.lua), {
+    pMap(scripts, (script) => _this.redis.script("load", script.lua), {
       concurrency: 10,
-    }).then(function () {
-      for (let i = 0; i < scripts.length; i++) {
-        _this.redis._addedScriptHashes[scripts[i].sha] = true;
-      }
-      return execPipeline();
-    });
+    })
+      .then(function () {
+        for (let i = 0; i < scripts.length; i++) {
+          _this.redis._addedScriptHashes[scripts[i].sha] = true;
+        }
+      })
+      .then(execPipeline, execPipeline);
```

Comment:
I think this will swallow errors thrown from `pMap` (as it doesn't work exactly as `.finally(execPipeline))`.

Should it be `.then(execPipeline, this.reject);`?